### PR TITLE
Update go to 1.23

### PR DESCRIPTION
**Pull request by ade18**
Update go to 1.21 to remediate crypto/tls vulnerability

### CODE REVIEW

**Comment by ade18** [comment]
https://github.com/IBM-Cloud/redli/issues/58

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19-alpine as builder
+FROM golang:1.21-alpine as builder
```
Comment (SANJEEV-Choubey) [inline Dockerfile:1]: It would be great if you could upgrade go version to 1.23

File: go.mod [hunk 2]
```diff
@@ -1,19 +1,19 @@
 module github.com/IBM-Cloud/redli
 
-go 1.19
+go 1.21
```
Comment (SANJEEV-Choubey) [inline go.mod:3]: It would be great if you could upgrade go version to 1.23

File: Dockerfile (hunk 1 above)
Comment (ade18) [reply Dockerfile]: Done

File: go.mod (hunk 2 above)
Comment (ade18) [reply go.mod]: Done

**Comment by ade18** [comment]
> @ade18 Thank you very much for your time and for creating this PR.Could below two things will be taken care?
> 
> * [ ]  Upgrade to Go 1.23
> * [ ]  Get rid of /vendor

Thank you for reviewing, i have made the changes. @SANJEEV-Choubey

**Comment by SANJEEV-Choubey** [comment]
Could you also get rid of vendor and start using the module?

**Comment by ade18** [comment]
> Could you also get rid of vendor and start using the module?
@SANJEEV-Choubey 
The vendor folder has been removed, its appearing like this in my commit because it has always been there in the repo , see https://github.com/IBM-Cloud/redli/tree/master/vendor . Please can you explain what you mean by  " using this module" ?

**Comment by SANJEEV-Choubey** [comment]
> > Could you also get rid of vendor and start using the module?
> > @SANJEEV-Choubey
> > The vendor folder has been removed, its appearing like this in my commit because it has always been there in the repo , see https://github.com/IBM-Cloud/redli/tree/master/vendor . Please can you explain what you mean by  " using this module" ?

@ade18 As we use go mod so now go/vendor is  not required, As you are updating go version it would be great we should only use modules in this PR otherwise a operate PR will be needed to remove vendor.

Action Item:- Delete vendor directory from this repo and Build and test it if it is working fine please update the PR and upload a test evidence.

**Comment by ade18** [comment]
> > > Could you also get rid of vendor and start using the module?
> > > @SANJEEV-Choubey
> > > The vendor folder has been removed, its appearing like this in my commit because it has always been there in the repo , see https://github.com/IBM-Cloud/redli/tree/master/vendor . Please can you explain what you mean by  " using this module" ?
> 
> @ade18 As we use go mod so now go/vendor is not required, As you are updating go version it would be great we should only use modules in this PR otherwise a operate PR will be needed to remove vendor.
> 
> Action Item:- Delete vendor directory from this repo and Build and test it if it is working fine please update the PR and upload a test evidence.

@SANJEEV-Choubey  the vendor directory is deleted in this PR thats why there is over 300+ files appearing being removed from the repository including the directory itself. You can navidate to my forked repo/branch and will see its indeed removed as shown in the PR. 

See evidences

**Compiling the code** 

Screenshot 2024-12-15 at 22 29 16


**See image being built:**

Screenshot 2024-12-15 at 22 32 28

**Review by SANJEEV-Choubey — APPROVED** [review]
LGTM

**Pull request merged**